lib/Word_Lib/Word_Lemmas.thy

```diff
@@ -1480,7 +1441,14 @@ lemma mask_shift_eq_mask_mask:
 lemma mask_shift_sum:
   "\<lbrakk> a \<ge> b; unat n = unat (p AND mask b) \<rbrakk>
    \<Longrightarrow> (p AND NOT(mask a)) + (p AND mask a >> b) * (1 << b) + n = (p :: 'a :: len word)"
-  by (metis and_not_mask mask_rshift_mult_eq_rshift_lshift mask_split_sum_twice word_unat.Rep_eqD)
+  apply (simp add: shiftl_def shiftr_def flip: push_bit_eq_mult take_bit_eq_mask word_unat_eq_iff)
+  apply (subst disjunctive_add)
+   apply (auto simp add: bit_simps)
+  apply (subst disjunctive_add)
+   apply (auto simp add: bit_simps)
+    apply (rule bit_word_eqI)
+  apply (auto simp add: bit_simps)
```

Comment:
this can't be your work, not sure why someone did this change

Reply:
Oh wow, that violates AFP rules as well. Need to fix there.